src/main/java/fiji/plugin/trackmate/gui/editor/LabkitLauncher.java

```diff
@@ -0,0 +1,394 @@
+import net.imglib2.type.numeric.integer.UnsignedShortType;
+import net.imglib2.view.Views;
+import sc.fiji.labkit.ui.LabkitFrame;
+import sc.fiji.labkit.ui.inputimage.DatasetInputImage;
+import sc.fiji.labkit.ui.labeling.Labeling;
+import sc.fiji.labkit.ui.models.DefaultSegmentationModel;
+import sc.fiji.labkit.ui.models.ImageLabelingModel;
+
+public class LabkitLauncher
+{
+
+	private final double[] calibration;
+
+	private final TrackMate trackmate;
+
+	private final EverythingDisablerAndReenabler disabler;
+
+	private ImgPlus< UnsignedShortType > previousIndexImg;
+
+	private int currentTimePoint;
+
+	private final boolean is2D;
+
+	private final double dt;
+
+	public LabkitLauncher( final TrackMate trackmate, final EverythingDisablerAndReenabler disabler )
+	{
+		this.trackmate = trackmate;
+		this.disabler = disabler;
+		final ImagePlus imp = trackmate.getSettings().imp;
+		this.calibration = TMUtils.getSpatialCalibration( imp );
+		this.is2D = DetectionUtils.is2D( imp );
+		this.dt = imp.getCalibration().frameInterval;
+	}
+
+	@SuppressWarnings( { "rawtypes", "unchecked" } )
+	protected void launch()
+	{
+		final ImagePlus imp = trackmate.getSettings().imp;
+		final ImgPlus src = TMUtils.rawWraps( imp );
+		final int timeAxis = src.dimensionIndex( Axes.TIME );
+
+		// Reslice for current time-point.
+		this.currentTimePoint = imp.getFrame() - 1;
+		final ImgPlus frame = ImgPlusViews.hyperSlice( src, timeAxis, currentTimePoint );
+		final ImpBdvShowable showable = ImpBdvShowable.fromImp( frame, imp );
+		final DatasetInputImage input = new DatasetInputImage( frame, showable );
+
+		// Prepare label image.
+		final AxisType[] axes = new AxisType[] { Axes.X, Axes.Y, Axes.Z, };
+		final long[] dims = new long[ is2D ? 2 : 3 ];
+		for ( int d = 0; d < dims.length; d++ )
+			dims[ d ] = src.dimension( src.dimensionIndex( axes[ d ] ) );
+		final Img< UnsignedShortType > lblImg = ArrayImgs.unsignedShorts( dims );
+		final double[] calibration = TMUtils.getSpatialCalibration( imp );
+		final ImgPlus< UnsignedShortType > lblImgPlus = new ImgPlus<>( lblImg, "LblImg", axes, calibration );
+
+		// Write spots in it with index = id + 1
+		final Iterable< Spot > spotsThisFrame = trackmate.getModel().getSpots().iterable( currentTimePoint, true );
+		for ( final Spot spot : spotsThisFrame )
+			spot.iterable( lblImgPlus ).forEach( p -> p.set( spot.ID() + 1 ) );
+
+		// Make a labeling model from it.
+		final Context context = TMUtils.getContext();
+		final DefaultSegmentationModel model = new DefaultSegmentationModel( context, input );
+		model.imageLabelingModel().labeling().set( Labeling.fromImg( lblImgPlus ) );
+
+		// Store a copy.
+		this.previousIndexImg = lblImgPlus.copy();
+
+		// Show LabKit.
+		final LabkitFrame labkit = LabkitFrame.show( model, "Edit TrackMate data frame " + ( currentTimePoint + 1 ) );
+		labkit.onCloseListeners().addListener( () -> reimportData( model.imageLabelingModel(), currentTimePoint ) );
```

Comment:
This will start a fully fledged Labkit with all features enabled. Your intention is to only edit that labels without all the extra features? Here is a demo on how to achieve this: [LabelingComponentDemo.java](https://github.com/juglab/labkit-ui/blob/master/src/test/java/demo/LabelingComponentDemo.java)

Reply:
Actually, I love the fully fledge Labkit. Do you have an opinion about what I should pick?